dotnet/src/dotnetframework/GxClasses/Services/ReflectionHelper.cs

```diff
@@ -221,6 +247,41 @@ internal static object[] ProcessParametersForInvoke(MethodInfo methodInfo, IDict
 			}
 			return parametersForInvocation;
 		}
+
+		internal static object[] ProcessParametersForInvoke(MethodInfo methodInfo, IList<object> parameters, IGxContext context = null)
+		{
+			var methodParameters = methodInfo.GetParameters();
+			object[] parametersForInvocation = new object[methodParameters.Length];
+			var idx = 0;
+			foreach (var methodParameter in methodParameters)
+			{
+				Type parmType = methodParameter.ParameterType;
+				if (IsByRefParameter(methodParameter))
+				{
+					parmType = parmType.GetElementType();
+				}
+				object value = parameters.ElementAt(idx);
+				if (!value.GetType().Equals(parmType))
+				{
+					//To avoid convertion from string type
+					if (value.GetType() != typeof(string))
+					{
+						var convertedValue = ConvertStringToNewType(value, parmType, context);
+						parametersForInvocation[idx] = convertedValue;
+					}
+					else
+					{
+						throw new ArgumentException("Type not match", methodParameter.Name);
```

Comment:
Minor suggestion for the message: "Type does not match" instead of "Type not match"